### PR TITLE
Add Lazy Library Loading

### PR DESCRIPTION
**Pull request by wenchuanchen**

### CODE REVIEW

**Comment by wenchuanchen** [comment]
doc this:

if(HIPBLAS_STATUS_INVALID_VALUE
   == hipblaslt_ext::getAlgosFromIndex(handle, algoIndex, testResults))
{
    std::cout << "Indexes are all out of bound." << std::endl;
    break;
}

File: library/src/amd_detail/rocblaslt/src/tensile_host.cpp [hunk 1]
```diff
@@ -2174,6 +2293,7 @@ rocblaslt_status isSolutionSupported(rocblaslt_handle            handle,
         }
 
         tensile_prob.setWorkspaceSize(algo->max_workspace_bytes);
+        // std::cout << "hardware pred = " << (*solution->hardwarePredicate)(*hardware) << std::endl;
```
Comment (wenchuanchen) [inline library/src/amd_detail/rocblaslt/src/tensile_host.cpp]: rm

File: library/src/amd_detail/rocblaslt/src/tensile_host.cpp [hunk 2]
```diff
@@ -2187,6 +2307,7 @@ rocblaslt_status isSolutionSupported(rocblaslt_handle            handle,
             log_error(__func__, "Solution is not supported");
             return rocblaslt_status_invalid_value;
         }
+        // std::cout << "prob pred = " << (*solution->problemPredicate)(tensile_prob) << std::endl;
```
Comment (wenchuanchen) [inline library/src/amd_detail/rocblaslt/src/tensile_host.cpp]: rm

File: tensilelite/Tensile/Source/lib/include/Tensile/MasterSolutionLibrary.hpp [hunk 3]
```diff
@@ -95,7 +95,7 @@ namespace Tensile
                                                                Hardware const&  hardware,
                                                                const int index) const override
         {
-            if(index >= solutions.size())
+            if(solutions.find(index) == solutions.end())
```
Comment (wenchuanchen) [inline tensilelite/Tensile/Source/lib/include/Tensile/MasterSolutionLibrary.hpp:98]: line 113 should be modified as well

**Review by KKyang — APPROVED** [review]

**Review by jichangjichang — COMMENTED** [review]
Please revise the descrition for getAlgosFromIndex API's return value section.

**Review by jichangjichang — APPROVED** [review]
lgtm

**Pull request merged**